### PR TITLE
Add GCF JVM language samples (Kotlin/Scala/Groovy)

### PR DESCRIPTION
**Pull request by ace-n**

### CODE REVIEW

File: functions/snippets/src/main/groovy/com/example/functions/GroovyHelloWorld.groovy [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import com.google.cloud.functions.HttpRequest
+import com.google.cloud.functions.HttpResponse
+
+class GroovyHelloWorld implements HttpFunction {
+    @Override
+    void service(HttpRequest request, HttpResponse response) {
+        response.writer.write("Hello World!")
+    }
+}
+// [END functions_helloworld_get]
```
Comment (lesv) [inline functions/snippets/src/main/groovy/com/example/functions/GroovyHelloWorld.groovy]: should there be a new line here?

File: functions/snippets/src/main/java/com/example/functions/ScalaHelloWorld.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.google.cloud.functions.{HttpFunction, HttpRequest, HttpResponse}
+
+// [START functions_helloworld_get]
+class ScalaHelloWorld extends HttpFunction {
+  override def service(httpRequest: HttpRequest, httpResponse: HttpResponse): Unit = {
+    httpResponse.getWriter.write("Hello World!")
+  }
+}
+// [END functions_helloworld_get]
```
Comment (lesv) [inline functions/snippets/src/main/java/com/example/functions/ScalaHelloWorld.scala]: newLine?

File: functions/snippets/src/main/kotlin/com/example/functions/KotlinHelloBackground.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            name = request.getFirstQueryParameter("name").get()
+        }
+        LOGGER.info(String.format("Hello %s!", name))
+    }
+
+    companion object {
+        private val LOGGER = Logger.getLogger(KotlinHelloBackground::class.java.name)
+    }
+}
+// [END functions_helloworld_background]
```
Comment (lesv) [inline functions/snippets/src/main/kotlin/com/example/functions/KotlinHelloBackground.kt]: newLine?

File: functions/snippets/src/main/kotlin/com/example/functions/KotlinHelloWorld.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.logging.Logger
+
+class KotlinHelloWorld : HttpFunction {
+    // Simple function to return "Hello World"
+    @Throws(IOException::class)
+    override fun service(request: HttpRequest, response: HttpResponse) {
+        response.writer.write("Hello World!")
+    }
+}
+// [END functions_helloworld_get]
```
Comment (lesv) [inline functions/snippets/src/main/kotlin/com/example/functions/KotlinHelloWorld.kt]: newLine?

File: functions/snippets/src/test/java/com/example/functions/GroovyTests.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  @After
+  public void afterTest() {
+    System.out.flush();
+    logHandler.flush();
+  }
+
+  @Test
+  public void groovyHelloWorldTest() throws IOException {
```
Comment (lesv) [inline functions/snippets/src/test/java/com/example/functions/GroovyTests.java]: Is this following our new naming standard for DRIFT tests?

File: functions/snippets/src/test/java/com/example/functions/GroovyTests.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    new GroovyHelloWorld().service(request, response);
+
+    writerOut.flush();
+    assertThat(responseOut.toString()).contains("Hello World!");
+  }
+
+  @Test
+  public void groovyHelloBackground_printsName() throws Exception {
```
Comment (lesv) [inline functions/snippets/src/test/java/com/example/functions/GroovyTests.java]: Does this follow the DRIFT naming of tests guide?

File: functions/snippets/src/test/java/com/example/functions/GroovyTests.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    when(request.getFirstQueryParameter("name")).thenReturn(Optional.of("John"));
+
+    new GroovyHelloBackground().accept(request, new MockContext());
+
+    String message = logHandler.getStoredLogRecords().get(0).getMessage();
+    assertThat("Hello John!").isEqualTo(message);
+  }
+
+  @Test
+  public void groovyHelloBackground_printsHelloWorld() throws Exception {
```
Comment (lesv) [inline functions/snippets/src/test/java/com/example/functions/GroovyTests.java]: Naming convention?

File: functions/snippets/src/test/java/com/example/functions/KotlinTests.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  @After
+  public void afterTest() {
+    System.out.flush();
+    logHandler.flush();
+  }
+
+  @Test
+  public void kotlinHelloWorldTest() throws IOException {
```
Comment (lesv) [inline functions/snippets/src/test/java/com/example/functions/KotlinTests.java]: Is this following our namingConvention?

File: functions/snippets/src/test/java/com/example/functions/ScalaTests.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  @After
+  public void afterTest() {
+    System.out.flush();
+    logHandler.flush();
+  }
+
+  @Test
+  public void scalaHelloWorldTest() throws IOException {
```
Comment (lesv) [inline functions/snippets/src/test/java/com/example/functions/ScalaTests.java]: namingConvention?

**Comment by ace-n** [comment]
@lesv naming convention has been addressed. LMK if you still want language-specific POMs.

(And per our earlier chat: there are indeed Kotlin files in this repo [in the `appengine` directory].)

**Pull request merged**